### PR TITLE
Add support for CCM* (with test vectors)

### PR DESCRIPTION
**Pull request by dgreen-arm**
## Description
Add support for CCM*. This PR is based on #1629 with the test vectors added.

## Status
**IN DEVELOPMENT**

## Requires Backporting

NO  

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated

### CODE REVIEW

**Comment by Patater** [comment]
The code review from #1629 approves this PR up to "CCM*: Remove superfluous braces" (4c579391b14f).

File: tests/suites/test_suite_ccm.function [hunk 1]
```diff
@@ -228,3 +228,140 @@ exit:
...
+                            int sec_level, char *add_hex,
+                            char *result_hex, int output_ret )
+{
+    unsigned char key[32];
+    unsigned char msg[50];
+    unsigned char iv[13];
+    unsigned char add[32];
+    unsigned char result[50];
+    unsigned char source_address[8];
+    unsigned char frame_counter[4];
```
Comment (Patater) [inline tests/suites/test_suite_ccm.function]: These are "magic number" sizes. It'd be nicer to have storage allocated based on the data present in the test vectors, as opposed to magic number arrays.

File: tests/suites/test_suite_ccm.function [hunk 2]
```diff
@@ -228,3 +228,140 @@ exit:
...
+    else
+        tag_len = 1 << ( sec_level % 4 + 1);
+
+    for( i = 0; i < source_address_len; i++ )
+        iv[i] = source_address[i];
+
+    for( i = 0; i < frame_counter_len; i++ )
+        iv[source_address_len + i] = frame_counter[i];
+
+    iv[source_address_len + frame_counter_len] = sec_level;
```
Comment (Patater) [inline tests/suites/test_suite_ccm.function]: It'd be nicer to have a struct for the iv type, so we don't have to manually do indexing of the elements like this.

**Comment by Patater** [comment]
retest

**Comment by Patater** [comment]
retest

**Comment by sbutcher-arm** [comment]
This PR can't both 'need review' and be 'ready for merge'. Which one is it?

cc: @Patater, @yanesca, @dgreen-arm

**Comment by yanesca** [comment]
It is ready for review. I have updated the label.

**Pull request merged**